scripts/sct_dmri_transpose_bvecs.py

```diff
@@ -24,66 +24,69 @@
 
 from __future__ import absolute_import, division
 
+import os
 import sys
-from msct_parser import Parser
+import argparse
+from spinalcordtoolbox.utils import Metavar, SmartFormatter
+
 from sct_utils import extract_fname, printv
 import sct_utils as sct
 
 
 # PARSER
 # ==========================================================================================
 def get_parser():
-    # parser initialisation
-    parser = Parser(__file__)
-
-    # # initialize parameters
-    # param = Param()
-    # param_default = Param()
-
     # Initialize the parser
-    parser = Parser(__file__)
-    parser.usage.set_description('Transpose bvecs file (if necessary) to get nx3 structure.')
-    parser.add_option(name='-bvec',
-                      type_value='file',
-                      description='Input bvecs file.',
-                      mandatory=True,
-                      example='bvecs.txt')
-    parser.add_option(name='-i',
-                      type_value='file',
-                      description='Input bvecs file.',
-                      mandatory=False,
-                      example='bvecs.txt',
-                      deprecated_by='-bvec')
-    parser.add_option(name='-o',
-                      type_value='file_output',
-                      description='Output bvecs file. By default input file is overwritten.',
-                      mandatory=False,
-                      example='bvecs_t.txt')
-    parser.add_option(name='-v',
-                      type_value='multiple_choice',
-                      description="""Verbose. 0: nothing. 1: basic. 2: extended.""",
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1', '2'])
+    parser = argparse.ArgumentParser(
+        description='Transpose bvecs file (if necessary) to get nx3 structure.',
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-bvec',
+        metavar=Metavar.file,
+        required=True,
+        help="Input bvecs file. Example: bvecs.txt"
+    )
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
```

Comment:
Is adding help as an optional argument here to make sure that it's added to the optional group? Because it's usually added to whole parser automatically I think.

Reply:
See https://github.com/neuropoly/spinalcordtoolbox/pull/2819#discussion_r476119649 for why we're doing this, I think.